ee/tabby-db/src/github_oauth_credential.rs

```diff
@@ -0,0 +1,123 @@
+use anyhow::Result;
+use chrono::{DateTime, Utc};
+use rusqlite::{params, OptionalExtension};
+
+use super::DbConn;
+
+pub struct GithubOAuthCredentialDAO {
+    pub id: i32,
+    pub client_id: String,
+    pub client_secret: String,
+    pub created_at: DateTime<Utc>,
+    pub updated_at: DateTime<Utc>,
+}
+
+impl GithubOAuthCredentialDAO {
+    fn from_row(row: &rusqlite::Row<'_>) -> std::result::Result<Self, rusqlite::Error> {
+        Ok(Self {
+            id: row.get(0)?,
+            client_id: row.get(1)?,
+            client_secret: row.get(2)?,
+            created_at: row.get(3)?,
+            updated_at: row.get(4)?,
+        })
+    }
+}
+
+/// db read/write operations for `github_oauth_credential` table
+impl DbConn {
+    pub async fn upsert_github_oauth_credential(
```

Comment:
```suggestion
    pub async fn update_github_oauth_credential(
```

I feel the semantic here can be restrict to single row operations